---
title: "paper: Nitrogen fixation by diverse diazotrophic communities can support population growth of arboreal ants"
author: "Maximilian Nepel"
date: "Feb 2022"
output: html_document
comment: "analyses of nifH gene and transcript sequencing. not all analyses ended up in the manuscript" 
---


# ###############################################################
# ###### analyses ###############################################
# ###############################################################

```{r, warning=FALSE, message=FALSE}
source("0_common.R", chdir = TRUE) # runs the common.R file
```

# loaded libraries
```{r}
all.pkgs.loaded <- gsub("package:","", search()[grep("package:",search())])
sessionInfo()$basePkgs
pkgs.loaded <- all.pkgs.loaded[!all.pkgs.loaded %in% sessionInfo()$basePkgs]

pkgs <- data.frame(matrix(ncol = 1,nrow = 0))
  colnames(pkgs) <- c("Version")
for (i in pkgs.loaded) {
  pkgs[i,1] <- sessionInfo()$otherPkgs[[i]]$Version
  }
print(pkgs)
```

# load subsetted phyloseq objects
```{r}
load(file = paste0(RD_DIR,"/ds.nifH.physeqs.2021-10-28.Robj")) # ds.nifH.physeqs.2021-07-19
```

## first glimpse on taxonomy
```{r}
physeq.norm.Fp.EPmean
physeq4norm.ansys <- physeq.norm.Fp.EPmean

# no. of OTUs (DNA only)
nrow(tax_table(physeq4norm.ansys))

# detected phyla
length(unique(tax_table(physeq4norm.ansys)[,"Phylum"]))
levels(as.factor(tax_table(physeq4norm.ansys)[,"Phylum"]))
```

### succession Fp vs. Ep - phylum & order stats
```{r}
physeq.norm.Fp.EPmean
physeq4norm.ansys <- physeq.norm.Fp.EPmean
plottitle <- "Fp & Ep (DNA, tree means)"
```

#### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz
Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")
```

#### 4.4.1 phylum taxonomy ------------------------------------------------------- manuscript (phylum stats)
```{r}
ab.Taxonomy <- Taxonomy #ab.Taxonomy.phylum
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Phylum, Type = ab.Taxonomy$Type, Marker = ab.Taxonomy$Marker, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
                       ) + scale_fill_manual(values=colPalette03
                       ) + scale_y_continuous(expand = c(0,1) # reduced space between bars and x-axis
                       ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
                       )
AllSample.bar + facet_wrap( ~ Type, scale="free", ncol=2) # rel. Abundance # + coord_flip()

levels(factor(tax.data2plot$Taxa))
mean.tax.Fp.EPmean <- aggregate(list(mean.abundance=tax.data2plot$Rel.Ab), by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
mean.tax.Fp.EPmean$round <- round(mean.tax.Fp.EPmean$mean.abundance, 1)
norm.Fp.EPmean.phylum.stats <- mean.tax.Fp.EPmean[order(mean.tax.Fp.EPmean$mean.abundance),]
norm.Fp.EPmean.phylum.stats

write.table(norm.Fp.EPmean.phylum.stats, 
              paste0(RESULTS_DIR,"/Fp.EPmean.phylum.stats",".", Sys.Date(),".txt"), 
              sep="\t", row.names=FALSE)
```

#### 4.4.3 order taxonomy - Fp & Ep mean - manuscript -------------------------in manuscript (Fig 2 & order stats)
```{r}
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.order
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))


# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
                       ) + scale_fill_manual(values=colPalette03
                       ) + scale_y_continuous(expand = c(0,1) # reduced space between bars and x-axis
                       ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
                       )
AllSample.bar + facet_wrap( ~ Type, scale="free", ncol=2) # rel. Abundance # + coord_flip()
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.EPmean.order",Sys.Date(),"png",sep = "."),width = 16, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.EPmean.order",Sys.Date(),"pdf",sep = "."),width = 16, height = 7, dpi = 300)

levels(factor(tax.data2plot$Taxa))
mean.tax.Fp.EPmean <- aggregate(list(mean.abundance=tax.data2plot$Rel.Ab), by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
mean.tax.Fp.EPmean$round <- round(mean.tax.Fp.EPmean$mean.abundance, 1)
norm.Fp.EPmean.order.stats <- mean.tax.Fp.EPmean[order(mean.tax.Fp.EPmean$mean.abundance),]
norm.Fp.EPmean.order.stats

write.table(norm.Fp.EPmean.order.stats,
              paste0(RESULTS_DIR,"/Fp.EPmean.order.stats",".", Sys.Date(),".txt"), 
              sep="\t", row.names=FALSE)
```

##### get order statistics ------------------------------- (order stats)
```{r}
# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.order.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)

for (j in levels(tax.data2plot$Type)){ #c("0.Fp", "2.Ep")
tax.data2plot.type <- tax.data2plot[tax.data2plot$Type == j,]
sum.tax.taxSample <- aggregate(list(rel.ab=tax.data2plot.type$Rel.Ab), by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0(j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
            paste(RESULTS_DIR,"/norm.",j,".order.stats.min.mean.max", Sys.Date(),".txt", sep = ""),
            sep="\t", row.names=FALSE)
}
```

#### 4.4.3 zCluster taxonomy - Fp & Ep mean - manuscript -------------------------in manuscript (Fig 4 & zClu stats)
```{r}
ab.Taxonomy.z.cluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Cluster", thresh = 0.2)

ab.Taxonomy <- ab.Taxonomy.z.cluster
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Zehr.Cluster, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
                       ) + scale_fill_manual(values=colPalette03
                       ) + scale_y_continuous(expand = c(0,1) # reduced space between bars and x-axis
                       ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
                       )
AllSample.bar + facet_wrap( ~ Type, scale="free", ncol=2) # rel. Abundance # + coord_flip()
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.EPmean.zClu",Sys.Date(),"png",sep = "."),width = 16, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.EPmean.zClu",Sys.Date(),"pdf",sep = "."),width = 16, height = 7, dpi = 300)

levels(factor(tax.data2plot$Taxa))
mean.tax.Fp.EPmean <- aggregate(list(mean.abundance=tax.data2plot$Rel.Ab), by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
mean.tax.Fp.EPmean$round <- round(mean.tax.Fp.EPmean$mean.abundance, 1)
norm.Fp.EPmean.zClu.stats <- mean.tax.Fp.EPmean[order(mean.tax.Fp.EPmean$mean.abundance),]
norm.Fp.EPmean.zClu.stats

write.table(norm.Fp.EPmean.order.stats,
              paste0(RESULTS_DIR,"/Fp.EPmean.zClu.stats",".", Sys.Date(),".txt"), 
              sep="\t", row.names=FALSE)
```

##### get zClu statistics ------------------------------- manuscript (cluster stats)
```{r}
# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.order.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)

for (j in levels(tax.data2plot$Type)){ #c("0.Fp", "2.Ep")
tax.data2plot.type <- tax.data2plot[tax.data2plot$Type == j,]
sum.tax.taxSample <- aggregate(list(rel.ab=tax.data2plot.type$Rel.Ab), by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0(j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
            paste(RESULTS_DIR,"/norm.",j,".zClu.stats.min.mean.max", Sys.Date(),".txt", sep = ""),
            sep="\t", row.names=FALSE)
}
```

## 0. core microbiome Fp & Ep (DNA, tree means)

### core() rel. ab. ------------in MS---------------------------------
```{r}
# merged Fp & Ep
physeq.norm.Fp.EPmean

(physeq.core <- physeq.norm.Fp.EPmean)
# (physeq.core <- prune_taxa(taxa_sums(physeq.core) > 0, physeq.core))

## OTU level
### OTUs present in 75% of samples
(sub.core <- core(physeq.core, detection = 0.0, prevalence = 75/100)) # detection=0, prev.=75/100: taxa need to have >0% of reads in >75% of samples
(tax.core <- as.data.frame(tax_table(sub.core)))
nrow(tax.core)

## genus level
(pseq.genus <- aggregate_taxa(physeq.core, "Genus"))
### genera present in 100% of samples
(sub.core <- core(pseq.genus, detection = 0.0, prevalence = 99/100))
(tax.core <- as.data.frame(tax_table(sub.core)))
nrow(tax.core)
```

## 1. alpha diversity - less diverse - Fp vs. Ep (DNA, tree means)

### succession Fp vs. Ep ----------------not in manuscript------------------------- alpha: sign.diff. Fp vs. Ep -----
```{r}
### Alpha diversitÃ¤t - phyloseq - not rarefy/transform beforehand!
physeq.abs.Fp.EPmean

(physeq.alphaPlot <- physeq.abs.Fp.EPmean)
  # (physeq.alphaPlot <- prune_taxa(taxa_sums(physeq.alphaPlot) > 0, physeq.alphaPlot))

plot_richness(physeq.alphaPlot, measures = c("Shannon"))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.alphaPlot), "data.frame")
alpha$Shannon <- estimate_richness(physeq.alphaPlot, measures = c("Shannon"))$Shannon

alpha$AntType <- paste0(alpha$Ant, "_", alpha$Type)
group_by(alpha, AntType) %>%
  summarise(
    count = n(),
    median = median(Shannon, na.rm = TRUE),
    IQR = IQR(Shannon, na.rm = TRUE)
  )

(alpha.res <- wilcox.test(Shannon ~ Type, data = alpha, exact = FALSE))
anova_result <- aov(Shannon ~ Type, alpha)
summary(anova_result)

plottitle <- "Shannon alpha diversity"

p <- ggplot(alpha, aes(reorder(Type), Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2
  ) + ggtitle(plottitle
  ) + coord_cartesian(ylim = c(0.3, 4.2)
  ) + geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
```

### alfari Fp vs. Ep -----------in MS------------------- alpha: sign.diff. ANTS x Fp vs. Ep -----
```{r}
physeq.abs.Fp.EPmean

physeq.alphaPlot <- subset_samples(physeq.abs.Fp.EPmean, Ant%in%("alf"))

plot_richness(physeq.alphaPlot, measures = c("Shannon"))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.alphaPlot), "data.frame")
alpha$Shannon <- estimate_richness(physeq.alphaPlot, measures = c("Shannon"))$Shannon

(alpha.res <- wilcox.test(Shannon ~ Type, data = alpha, exact = FALSE))

plottitle <- "Shannon alpha diversity - alfari"

p <- ggplot(alpha, aes(reorder(Type), Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2
  ) + ggtitle(plottitle
  ) + coord_cartesian(ylim = c(0.3, 4.2)
  ) + geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.alf.ylim",Sys.Date(),"png",sep = "."),width = 5, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.alf.ylim",Sys.Date(),"pdf",sep = "."),width = 5, height = 6, dpi = 300)
```

### constructor Fp vs. Ep --------in MS------------------- alpha: sign.diff. ANTS x Fp vs. Ep -----
```{r}
physeq.abs.Fp.EPmean

physeq.alphaPlot <- subset_samples(physeq.abs.Fp.EPmean, Ant%in%("con"))

plot_richness(physeq.alphaPlot, measures = c("Shannon"))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.alphaPlot), "data.frame")
alpha$Shannon <- estimate_richness(physeq.alphaPlot, measures = c("Shannon"))$Shannon

(alpha.res <- wilcox.test(Shannon ~ Type, data = alpha, exact = FALSE))

plottitle <- "Shannon alpha diversity - constr"

p <- ggplot(alpha, aes(reorder(Type), Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2
  ) + ggtitle(plottitle
  ) + coord_cartesian(ylim = c(0.3, 4.2)
  ) + geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.con.ylim",Sys.Date(),"png",sep = "."),width = 5, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.con.ylim",Sys.Date(),"pdf",sep = "."),width = 5, height = 6, dpi = 300)
```

### xanthochroa Fp  -------------in MS-------------------
```{r}
physeq.abs.Fp.EPmean

physeq.alphaPlot <- subset_samples(physeq.abs.Fp.EPmean, Ant%in%("xan"))

plot_richness(physeq.alphaPlot, measures = c("Shannon"))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.alphaPlot), "data.frame")
alpha$Shannon <- estimate_richness(physeq.alphaPlot, measures = c("Shannon"))$Shannon

mean(alpha[grep("0.Fp", alpha$Type),]$Shannon)

plottitle <- "Shannon alpha diversity - xan"

p <- ggplot(alpha, aes(reorder(Type), Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2
  ) + ggtitle(plottitle
  ) + coord_cartesian(ylim = c(0.3, 4.2)
  ) + geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.xan.ylim",Sys.Date(),"png",sep = "."),width = 2.5, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.EPmean.xan.ylim",Sys.Date(),"pdf",sep = "."),width = 2.5, height = 6, dpi = 300)
```

## 4. beta diversity - succession - Fp vs. Ep (DNA, tree means) ------------------------- type/ant: sign. R2: 0.081/0.053 -----

### sucession Fp vs. Ep (DNA, tree means)
```{r}
physeq.norm.Fp.EPmean
physeq4norm.ansys <- physeq.norm.Fp.EPmean
physeq4abs.ansys <- physeq.abs.Fp.EPmean

plottitle <- "succession - Fp & Ep (DNA, tree means)"
```

#### 4.1 PERMANOVA - beta diversity correlating with patch type & ant species?
```{r}
(m.Fp.EPmean.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type, 
                        data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Fp.EPmean.Ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant,
                       data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Fp.EPmean.TypeAnt <- adonis(otu_table(physeq4norm.ansys) ~ Type*Ant, 
                           data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Fp.EPmean.AntType <- adonis(otu_table(physeq4norm.ansys) ~ Ant*Type, 
                           data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

#### 4.2 PCoA ----------------------not in MS--------------------------
```{r}
pcoa.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type & Type
p <- PlotOrd.Fig3a(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.EPmean.Type",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.EPmean.Type",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)

# Ant
p <- PlotOrd.Fig3a(dbRDA.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.EPmean.Ant",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.EPmean.Ant",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)

# Type & Ant
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
```

#### 4.3 dbRDA ---------------- in manuscript (Fig. 3a) -------------------------
```{r}
dbRDA.obj <- capscale(otu_table(physeq4norm.ansys) ~ Type, data=get_variable(physeq4norm.ansys),  distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.Fig3a(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Fp.EPmean.Type",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Fp.EPmean.Type",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)
```

### alfari Fp vs. Ep
```{r}
physeq4norm.ansys <- subset_samples(physeq.abs.Fp.EPmean, Ant%in%("alf"))

plottitle <- "beta div - Fp&Ep - ONLY alf"

#### adonis
(m.Fp.EPmean.alf.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type, 
                        data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))

#### PCoA
pcoa.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type & Type
p <- PlotOrd.Fig3a(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
```

### constructor Fp vs. Ep
```{r}
physeq4norm.ansys <- subset_samples(physeq.abs.Fp.EPmean, Ant%in%("con"))

plottitle <- "beta div - Fp&Ep - ONLY constructor"

#### adonis
(m.Fp.EPmean.con.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type, 
                        data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))

#### PCoA
pcoa.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type & Type
p <- PlotOrd.Fig3a(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
```

## 2.1 EPs of different age - Ep (DNA, only trees > 1 sample) ----- age: not sign. / tree R2: 0.777 -----
```{r}
physeq.norm.Ep.age

physeq4norm.ansys <- physeq.norm.Ep.age

plottitle <- "EPs of different age - Ep - age"
```

### adonis
```{r}
(m.Ep.age.Age <- adonis(otu_table(physeq4norm.ansys) ~ Age, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Ep.age.Age.strataAnt <- adonis(otu_table(physeq4norm.ansys) ~ Age, data = get_variable(physeq4norm.ansys), strata=get_variable(physeq4norm.ansys)$Ant, method = "bray", permutations = 9999))
(m.Ep.age.Tree <- adonis(otu_table(physeq4norm.ansys) ~ Tree, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

### PCoA
```{r}
pcoa.obj <- capscale((otu_table(physeq4norm.ansys)) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Age & Type
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Age", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# Tree
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Ep.age.Tree",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Ep.age.Tree",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)
```

### dbRDA
```{r}
dbRDA.obj <- capscale((otu_table(physeq4norm.ansys)) ~ Tree, data=get_variable(physeq4norm.ansys), distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Tree
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)
  ) + ggtitle(plottitle
  ) + xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")
  ) + ylab(paste("dbRDA 2 (", explained[2], "%)", sep = "")
  # ) + geom_text(aes(label=Tree),hjust=0, vjust=0
)

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Ep.age.LabelTree7.5",Sys.Date(),"png",sep = "."),width = 7.5, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Ep.age.LabelTree7.5",Sys.Date(),"pdf",sep = "."),width = 7.5, height = 6, dpi = 300)
```

## 2.2 tree a significant factor??? IP (only trees > 1 sample) ------------- tree: not sign. -----
```{r}
physeq.norm.Fp.age

physeq4norm.ansys <- physeq.norm.Fp.age

plottitle <- "plant-assiciated community? - Fp - tree"
```

### adonis
```{r}
(m.Fp.age.Tree <- adonis(otu_table(physeq4norm.ansys) ~ Tree, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Fp.age.AntTree <- adonis(otu_table(physeq4norm.ansys) ~ Ant*Tree, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.Fp.age.strataTree <- adonis(otu_table(physeq4norm.ansys) ~ Tree, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999, strata=get_variable(physeq4norm.ansys)$Ant))
```

### PCoA
```{r}
pcoa.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Tree
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.age.Tree",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.age.Tree",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)

# Tree & Ant
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Tree", treatment = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.age.TreeAnt",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.age.TreeAnt",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)
```

## 3. role ant - Fp | Ep (DNA, tree means)

### 3.1 Fp -------------------------------not in manuscript--------------------------- ant: not sign. -----
```{r}
physeq.norm.Fp
physeq4norm.ansys <- physeq.norm.Fp

plottitle <- "role ant - Fp"
```

#### adonis
```{r}
(m.Fp.ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

#### PCoA
```{r}
pcoa.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Ant & Type
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.AntType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.AntType",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)
```

#### 3.1.1 genus taxonomy - Fp - checking for revision
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz
Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")

ab.Taxonomy.tax <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.tax
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
                       ) + scale_fill_manual(values=colPalette03
                       ) + scale_y_continuous(expand = c(0,1) # reduced space between bars and x-axis
                       ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
                       )
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.genus",Sys.Date(),"png",sep = "."),width = 16, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.genus",Sys.Date(),"pdf",sep = "."),width = 16, height = 7, dpi = 300)
```

### 3.3 Ep --------------------------------------------------------------------- ant: sign. R2: 0.043 -----
```{r}
physeq.norm.Ep.mean
physeq4norm.ansys <- physeq.norm.Ep.mean
  # (physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "Ep (DNA, tree means)"
```

#### 3.3.1 adonis ----------------in MS-----------------------
```{r}
(m.Ep.mean.ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

#### 3.3.4 dbRDA -----------------in MS---------------------------------------- in manuscript (Fig. 3c) ----------
```{r}
dbRDA.obj <- capscale(otu_table(physeq4norm.ansys) ~ Ant, data=get_variable(physeq4norm.ansys), distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Marker = get_variable(physeq4norm.ansys)$Marker, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Ant & Type
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)
  ) + ggtitle(plottitle
  ) + xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")
  ) + ylab(paste("dbRDA 2 (", explained[2], "%)", sep = "")
)

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Ep.mean.AntType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Ep.mean.AntType",Sys.Date(),"pdf",sep = "."),width = 7, height = 6, dpi = 300)
```

#### 3.3.5 tax.bar.plot

##### base for stacked bar charts & heatmaps
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz
Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")
```

##### order taxonomy - Ep DNA alf vs. con ---------removed from MS-------------------------- summary manuscript (Fig 5)
```{r}
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)

# select target samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Ant), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within every ant species
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
AllSample.bar # + facet_wrap( ~ Ant, scale="free", ncol=2) # rel. Abundance 

ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.order.reorder",Sys.Date(),"png",sep = "."),width = 16, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.order.reorder",Sys.Date(),"pdf",sep = "."),width = 16, height = 7, dpi = 300)

## plot stacked bar chart per Ant
Ant.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
Ant.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.order.sumAnt",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.order.sumAnt",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

###### get statistics - order taxonomy ---------description questionable if it will stay in MS----------
```{r}
  # Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa","alf.rel.ab","con.rel.ab", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7, 
              paste(RESULTS_DIR,"/EPmean.alf.con.mean.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation within ant species
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
sum.tax.taxSample

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0("EPmean.",j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
              paste(RESULTS_DIR,"/EPmean.",j,".min.mean.max.order.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

##### zehr.cluster taxonomy -----------------------plot was cut------------------------------ summary manuscript (Fig 5)
```{r}
ab.Taxonomy.z.cluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Cluster", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.z.cluster
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Zehr.Cluster, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within every ant species
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=2) # rel. Abundance 

ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.zClu.reorder",Sys.Date(),"png",sep = "."),width = 16, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.zClu.reorder",Sys.Date(),"pdf",sep = "."),width = 16, height = 7, dpi = 300)

  ## plot stacked bar chart per Type
Ant.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
Ant.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.zClu.ant",Sys.Date(),"png",sep = "."),width = 6, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EPmean.zClu.ant",Sys.Date(),"pdf",sep = "."),width = 6, height = 6, dpi = 300)
```

###### get statistics - zehr.cluster ---------------description still in MS----------------------------
```{r}
# sample variation within ant species
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

sum.tax.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0("EPmean.Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6,
              paste(RESULTS_DIR,"/EPmean.min.mean.max.zClu.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa","alf.rel.ab","con.rel.ab", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7, 
              paste(RESULTS_DIR,"/EPmean.alf.con.mean.zClu.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation within ant species
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
sum.tax.taxSample

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0("EPmean.",j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
              paste(RESULTS_DIR,"/EPmean.",j,".min.mean.max.zClu.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

## 5. acitve diazos - Ep (only RNA + resp. DNA)
```{r}
# in how many colonies could we detect nifH transcript?
(act <- nlevels(get_variable(physeq.norm.activity)$Tree)) # no. of E colonies transcript detected
(Ecol <- nlevels(get_variable(physeq.norm.Ep.mean)$Tree)) # no. of investigated E colonies
act/Ecol*100 # % of Ecol in which transcript could be detected
```

### alpha diversity - shannon - all activity EP.mean
```{r}
physeq.abs.activity
physeq4abs.ansys <- physeq.abs.activity
  # (physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))

# testing Ep alpha diversity for significance
alpha <- as(sample_data(physeq4abs.ansys), "data.frame")
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon
alpha$AntMarker <- paste0(alpha$Ant,".",alpha$Marker)

group_by(alpha, AntMarker) %>%
  summarise(
    count = n(),
    median = median(Shannon, na.rm = TRUE),
    IQR = IQR(Shannon, na.rm = TRUE)
  )

plottitle <- "Shannon alpha diversity - activity"

p <- ggplot(alpha, aes(AntMarker, Shannon))
p + geom_boxplot(aes(fill = Marker), outlier.colour="red", outlier.shape=NA, outlier.size=2
  ) + ggtitle(plottitle
  ) + coord_cartesian(ylim = c(0.3, 4.2)
  ) + geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.activity.AntMarker.ylim.dots",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.activity.AntMarker.ylim.dots",Sys.Date(),"pdf",sep = "."),width = 5, height = 4, dpi = 300)
```

#### alpha diversity - shannon - ONLY alfari
```{r}
(physeq.abs.act.alf <- subset_samples(physeq4abs.ansys, Ant%in%("alf")))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.abs.act.alf), "data.frame")
alpha$Shannon <- estimate_richness(physeq.abs.act.alf, measures = c("Shannon"))$Shannon

(alpha.res <- wilcox.test(Shannon ~ Marker, data = alpha, exact = FALSE))
```

#### alpha diversity - shannon - ONLY constructor
```{r}
(physeq.abs.act.con <- subset_samples(physeq4abs.ansys, Ant%in%("con")))

# testing Fp vs. Ep alpha diversity for significance
alpha <- as(sample_data(physeq.abs.act.con), "data.frame")
alpha$Shannon <- estimate_richness(physeq.abs.act.con, measures = c("Shannon"))$Shannon

(alpha.res <- wilcox.test(Shannon ~ Marker, data = alpha, exact = FALSE))
```

### beta diversity
```{r}
physeq.norm.activity
physeq4norm.ansys <- physeq.norm.activity
plottitle <- "acitve diazos - Ep (only RNA + resp. DNA)"
```

### adonis
```{r}
(m.Ep.active.Marker <- adonis(otu_table(physeq.norm.activity) ~ Marker, data = get_variable(physeq.norm.activity), method = "bray", permutations = 9999))
(m.Ep.active.Marker.stratoAnt <- adonis(otu_table(physeq.norm.activity) ~ Marker, data = get_variable(physeq.norm.activity), strata=get_variable(physeq.norm.activity)$Ant, method = "bray", permutations = 9999))
```

### PCoA
```{r}
pcoa.obj <- capscale(otu_table(physeq.norm.activity) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq.norm.activity)$Name, Type = get_variable(physeq.norm.activity)$Type, Marker = get_variable(physeq.norm.activity)$Marker, Year = get_variable(physeq.norm.activity)$Year, Ant = get_variable(physeq.norm.activity)$Ant, Age = get_variable(physeq.norm.activity)$Age, Tree = get_variable(physeq.norm.activity)$Tree)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Marker
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Marker", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# Marker & Ant
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "Marker", treatment = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
```

### base for stacked bar charts & heatmaps
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq.norm.activity)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)

Taxonomy2 <- cbind(physeq.norm.activity@tax_table@.Data, OTU4tax)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz
Taxonomy <- merge(Taxonomy2, get_variable(physeq.norm.activity), by="Name")
```

#### order taxonomy ----------------------in MS---------------------------------- Fig. 5 bottom-----
```{r}
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within ant species
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within DNA/RNA
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Marker == tax.data2plot$Marker[i]])
  tax.data2plot$Marker.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
AllSample.bar + facet_wrap( ~ Marker, scale="free", ncol=1)

ggsave(paste(PLOTS_DIR,"/bar.tax.Ep.active.order.sort",Sys.Date(),"png",sep = "."),width = 14, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Ep.active.order.sort",Sys.Date(),"pdf",sep = "."),width = 14, height = 7, dpi = 300)

AllSample.bar + facet_wrap( ~ Marker*Ant, scale="free", ncol=2) # rel. Abundance

  ## plot stacked bar chart per Marker
Marker.bar <- qplot(x=Marker,data=tax.data2plot,geom="bar",weight=Marker.Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
Marker.bar + facet_wrap( ~ Ant, scale="free") # rel. Abundance
```

###### get statistics - order taxonomy ---------description questionable if it will stay in MS----------
```{r}
# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt 
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa.mean.rel.ab","alf","con", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/activity.alf.con.mean.order.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within ant species
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

for (i in c("DNA", "RNA")){
tax.data2plot.marker <- tax.data2plot[tax.data2plot$Marker == i,]

for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot.marker[tax.data2plot.marker$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
sum.tax.taxSample

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0("activity.",i,".",j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
              paste(RESULTS_DIR,"/activity.",i,".",j,".min.mean.max.order.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
}
```

#### zehr.cluster taxonomy ------------------------------------------------------ supplementary
```{r}
ab.Taxonomy.z.cluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Cluster", thresh = 0.2)
ab.Taxonomy <- ab.Taxonomy.z.cluster

tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Zehr.Cluster, Type = ab.Taxonomy$Type, Year = ab.Taxonomy$Year, Ant = ab.Taxonomy$Ant, Marker = ab.Taxonomy$Marker, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$Tree)
tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Year", "Ant", "Marker", "Age", "Tree"), summarise, Freq = sum(Freq))

# for stacked bar charts
  ## adds rel.ab. for every OTU per sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within ant species
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## adds rel.ab. for every OTU within DNA/RNA
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Marker == tax.data2plot$Marker[i]])
  tax.data2plot$Marker.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
AllSample.bar + facet_wrap( ~ Marker, scale="free", ncol=1) # rel. Abundance 

ggsave(paste(PLOTS_DIR,"/bar.tax.activity.zClu",Sys.Date(),"png",sep = "."),width = 10, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.activity.zClu",Sys.Date(),"pdf",sep = "."),width = 10, height = 7, dpi = 300)

AllSample.bar + facet_wrap( ~ Marker*Ant, scale="free") # rel. Abundance 

  ## plot stacked bar chart per Marker
Marker.bar <- qplot(x=Marker,data=tax.data2plot,geom="bar",weight=Marker.Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
Marker.bar + facet_wrap( ~ Ant, scale="free") # rel. Abundance

  ## plot stacked bar chart per Ant
Ant.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle
  ) + scale_fill_manual(values=colPalette03
  ) + scale_y_continuous(expand = c(0,1)
  ) + theme(axis.text.x = element_text(angle = 30, hjust = 1)
)
Ant.bar + facet_wrap( ~ Marker, scale="free")

# ggsave(paste(PLOTS_DIR,"/bar.tax.activity.zClu.Ant",Sys.Date(),"png",sep = "."),width = 6, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/bar.tax.activity.zClu.Ant",Sys.Date(),"pdf",sep = "."),width = 6, height = 6, dpi = 300)
```

##### get cluster statistics -------------------------------------------------------- supplementary
```{r}
# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt 
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa.mean.rel.ab","alf","con", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/activity.alf.con.mean.zClu.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within ant species
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

for (i in c("DNA", "RNA")){
tax.data2plot.marker <- tax.data2plot[tax.data2plot$Marker == i,]

for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot.marker[tax.data2plot.marker$Ant == j,]
# tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
sum.tax.taxSample

# get statistics min mean max per ant species
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2$Taxa <- factor(sum.tax.min.max.2$Taxa)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0("activity.",i,".",j,".Taxa"),"min.rel.ab","mean.rel.ab", "max.rel.ab") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8,
              paste(RESULTS_DIR,"/activity.",i,".",j,".min.mean.max.zClu.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
}
```












# END